packages/core/src/stringifyPolicy.js

```diff
@@ -1,11 +1,25 @@
 const jsonStringify = require('json-stable-stringify')
 
+/**
+ * @import {LavaMoatPolicy, LavaMoatPolicyOverrides, LavaMoatPolicyDebug} from './schema'
+ */
+
 module.exports = {
+  /**
+   * Stringifies a policy, sorting the keys in a diff-friendly way.
+   *
+   * @param {LavaMoatPolicy | LavaMoatPolicyOverrides | LavaMoatPolicyDebug} policy
+   * @returns {string}
+   */
   jsonStringifySortedPolicy(policy) {
-    return jsonStringify(policy, {
+    const result = jsonStringify(policy, {
       space: 2,
       cmp: diffFriendlyPolicyKeysComparator,
     })
+    if (result === undefined) {
```

Comment:
nothing says typescript was being annoying like an undefined check on something that can't reasonably be undefined :sweat_smile:

Reply:
_au contraire_

```js
import stringify from 'json-stable-stringify'

const foo = stringify()

foo === undefined // true
```

it's just that the types aren't narrow enough; it's typed as returning `string | undefined`, and it should actually be overloaded so that if `undefined` extends parameter `T`, then the return type will be `undefined` (and `string` otherwise).

I guess it's there on DT and I could send a PR.